src/main/java/com/google/sps/data/Folder.java

```diff
@@ -44,23 +52,16 @@ public Entity createEntity(Key userKey) {
         // Set owner of folder 
         Entity folder = new Entity("Folder", userKey);
 
+        // Store initial folder entity without properties to generate auto ID for entity
+        // This is necessary to obtain a "complete" key
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(folder);
+
         folder.setProperty("folderName", this.folderName);
         folder.setProperty("folderDefaultLanguage", this.folderDefaultLanguage);
         folder.setProperty("folderKey", KeyFactory.keyToString(folder.getKey()));
```

Comment:
What do you mean exactly? Do you mean why I am storing the folder's key as a property?

Reply:
Isn't the "complete key" already stored in the entity by default — you can call `getKey()`?

Reply:
Unless I am creating an entity where I manually identify with a unique ID, I can call getKey(). Otherwise, if I am just creating an entity e.g, new Entity("Folder", ancestry key) -- this would still be an incomplete key. That's why I'd need datastore to generate an ID for the entity by calling the put(). I think getKey() requires the entity to have a complete key otherwise it'd throw an error of "incomplete key."

I like your idea though. It is much simpler to call entity.getKey() when I am querying for all the entities from datastore and calling the class method, setKey() in the Folder object during the doGet method.